### PR TITLE
Home work user

### PR DESCRIPTION
**Pull request by viklink**

### CODE REVIEW

File: src/source/it/butov/homeTask/FractionNumberImpl.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+package source.it.butov.homeTask;
+
+import java.util.ArrayList;
```
Comment (ViacheslavButov) [inline src/source/it/butov/homeTask/FractionNumberImpl.java]: Remove unused imports

File: src/source/it/butov/homeTask/FractionNumberOperationImpl.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private int divisorB;
+
+    FractionNumber a =  new FractionNumberImpl (dividentA, divisorA);
+
+    FractionNumber b = new FractionNumberImpl (dividentB,divisorB);
+
+	@Override
+	public FractionNumber add(FractionNumber a, FractionNumber b) {
+		
+		int dividentC = a.getDivident() * b.getDivisor() + b.getDivident() + a.getDivisor();
```
Comment (ViacheslavButov) [inline src/source/it/butov/homeTask/FractionNumberOperationImpl.java:20]: Do not forget about code formatting. Do not leave blank lines

File: src/source/it/butov/user/UserRetrieverImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.io.File;
+import java.util.Scanner;
+
+public class UserRetrieverImpl implements UserRetriever {
+
+	private UserConverter userConverter;
+	private String csvString;
+
+	public UserRetrieverImpl(UserConverter userConverter) {
```
Comment (ViacheslavButov) [inline src/source/it/butov/user/UserRetrieverImpl.java:13]: You are require 'userConverter' but do not assign it to class field

File: src/source/it/butov/user/UserRetrieverImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+	@Override
+	public User[] retrieve() throws Exception {
+
+		// User[] users = null;
+
+		Scanner scanner = new Scanner(new File("hometask.txt"));
+
+		while (scanner.hasNextLine()) {
+			String line = scanner.nextLine().trim();
```
Comment (ViacheslavButov) [inline src/source/it/butov/user/UserRetrieverImpl.java]: you are retrieving string from file and then do nothing...

File: src/source/it/butov/user/UserRetrieverImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package source.it.butov.user;
```
Comment (ViacheslavButov) [inline src/source/it/butov/user/UserRetrieverImpl.java:1]: The class will not work properly

File: src/source/it/butov/user/UserServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+public class UserServiceImpl implements UserService {
+
+	// private User[] users;
+	ErrorType result;
+	UserRetriever userRetriever;
+	UserValidator userValidator;
+
+	public UserServiceImpl(UserRetriever userRetriever, UserValidator userValidator) {
+
```
Comment (ViacheslavButov) [inline src/source/it/butov/user/UserServiceImpl.java]: The same as in UserRetrieverImpl class

File: src/source/it/butov/user/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+	}
+
+	// User user = userRetriever.retrieve();
+
+	@Override
+	public User[] getAllUsers() throws Exception {
+
+		User user = new User();
```
Comment (ViacheslavButov) [inline src/source/it/butov/user/UserServiceImpl.java]: this method should use user retriever and that's it

File: src/source/it/butov/homeTask/FractionNumberImpl.java (hunk 1 above)
Comment (viklink) [reply src/source/it/butov/homeTask/FractionNumberImpl.java]: @ViacheslavButov done

File: src/source/it/butov/homeTask/FractionNumberOperationImpl.java (hunk 2 above)
Comment (viklink) [reply src/source/it/butov/homeTask/FractionNumberOperationImpl.java:20]: @ViacheslavButov done

File: src/source/it/butov/user/UserRetrieverImpl.java (hunk 3 above)
Comment (viklink) [reply src/source/it/butov/user/UserRetrieverImpl.java:13]: @ViacheslavButov done

File: src/source/it/butov/user/UserRetrieverImpl.java (hunk 4 above)
Comment (viklink) [reply src/source/it/butov/user/UserRetrieverImpl.java]: @ViacheslavButov done

File: src/source/it/butov/user/UserRetrieverImpl.java (hunk 5 above)
Comment (viklink) [reply src/source/it/butov/user/UserRetrieverImpl.java:1]: @ViacheslavButov done

File: src/source/it/butov/user/UserServiceImpl.java (hunk 6 above)
Comment (viklink) [reply src/source/it/butov/user/UserServiceImpl.java]: @ViacheslavButov done

File: src/source/it/butov/user/UserServiceImpl.java (hunk 7 above)
Comment (viklink) [reply src/source/it/butov/user/UserServiceImpl.java]: @ViacheslavButov done

**Pull request merged**